roles/tests-integration/tests/common/mod.rs

```diff
@@ -0,0 +1,152 @@
+use bitcoind::{bitcoincore_rpc::RpcApi, BitcoinD, Conf};
+use flate2::read::GzDecoder;
+use std::{
+    env,
+    fs::{create_dir_all, File},
+    io::{BufReader, Read},
+    path::{Path, PathBuf},
+};
+use tar::Archive;
+
+const VERSION_TP: &str = "0.1.7";
+
+fn download_bitcoind_tarball(download_url: &str) -> Vec<u8> {
+    let response = minreq::get(download_url)
+        .send()
+        .expect(&format!("Cannot reach URL: {}", download_url));
+    assert_eq!(
+        response.status_code, 200,
+        "URL {} didn't return 200",
+        download_url
+    );
+    response.as_bytes().to_vec()
+}
+
+fn read_tarball_from_file(path: &str) -> Vec<u8> {
+    let file = File::open(path).expect(&format!(
+        "Cannot find {:?} specified with env var BITCOIND_TARBALL_FILE",
+        path
+    ));
+    let mut reader = BufReader::new(file);
+    let mut buffer = Vec::new();
+    reader.read_to_end(&mut buffer).unwrap();
+    buffer
+}
+
+fn unpack_tarball(tarball_bytes: &[u8], destination: &Path) {
+    let decoder = GzDecoder::new(tarball_bytes);
+    let mut archive = Archive::new(decoder);
+    for mut entry in archive.entries().unwrap().flatten() {
+        if let Ok(file) = entry.path() {
+            if file.ends_with("bitcoind") {
+                entry.unpack_in(destination).unwrap();
+            }
+        }
+    }
+}
+
+fn get_bitcoind_filename(os: &str, arch: &str) -> String {
+    match (os, arch) {
```

Comment:
Should we move these to a separate utils module?

Reply:
That was my initial though as well. But I think we should wait a bit and see if those functions are gonna be used by other roles, otherwise they really should be part of the `TemplateProvier` module in `tests::common::mod::TemplateProvider`.

Reply:
Yeah, that makes sense. I don't think these utility methods will be used anywhere else except here. Let's keep them here for now, and if we find that another module uses them, we can move them to a separate utility module.